src/main/java/seedu/duke/logic/similaritycheck/SimilarityCheck.java

```diff
@@ -21,33 +21,32 @@
 
     private static double getSimilarity(String givenLoc, String inputLoc) {
         if (givenLoc.length() > inputLoc.length()) {
-            return (givenLoc.length() - editDistance(givenLoc, inputLoc)) / (double) givenLoc.length();
+            return 1 - editDistance(givenLoc, inputLoc) / (double) givenLoc.length();
         }
 
-        return (inputLoc.length() - editDistance(inputLoc, givenLoc)) / (double) inputLoc.length();
+        return 1 - editDistance(inputLoc, givenLoc) / (double) inputLoc.length();
     }
 
     //Source: http://rosettacode.org/wiki/Levenshtein_distance#Java
-    private static int editDistance(String a, String b) {
-        a = a.toLowerCase();
-        b = b.toLowerCase();
-        // i == 0
-        int [] costs = new int [b.length() + 1];
+    private static int editDistance(String loc1, String loc2) {
+        loc1 = loc1.toLowerCase();
+        loc2 = loc2.toLowerCase();
+
+        int [] costs = new int [loc2.length() + 1];
         for (int j = 0; j < costs.length; j++) {
             costs[j] = j;
         }
-        for (int i = 1; i <= a.length(); i++) {
-            // j == 0; nw = lev(i - 1, j)
+        for (int i = 1; i <= loc1.length(); i++) {
             costs[0] = i;
-            int nw = i - 1;
-            for (int j = 1; j <= b.length(); j++) {
-                int cj = Math.min(1 + Math.min(costs[j], costs[j - 1]),
-                        a.charAt(i - 1) == b.charAt(j - 1) ? nw : nw + 1);
-                nw = costs[j];
-                costs[j] = cj;
+            int before = i - 1;
+            for (int j = 1; j <= loc2.length(); j++) {
+                int now = Math.min(1 + Math.min(costs[j], costs[j - 1]),
+                        loc1.charAt(i - 1) == loc2.charAt(j - 1) ? before : before + 1);
+                before = costs[j];
+                costs[j] = now;
             }
         }
-        return costs[b.length()];
+        return costs[loc2.length()];
     }
 
```

Comment:
Good job changing the variables as it makes it more readable. MIght be good to add a Javadoc to further explain it.

Reply:
Yep! doing that in a different pull request